### PR TITLE
[ENH] faster pretrained text encoder

### PR DESCRIPTION
**Pull request by paxcema**
Note: this builds on top of #1145 so the PR is set against that branch. However, it should be merged into `staging` eventually!

## Changelog
Refactored `PretrainedTextEncoder` training loop to introduce a few things that could help with getting a better runtime-accuracy tradeoff by default:
1. Create and use a validation split.
2. Early stopping strategies based on validation and training loss.
3. Replaced a few operations for vectorized equivalents.

## Effect
Tests indicate a further 15% improvement compared to #1145 in text classification tasks that use the UnitMixer

### CODE REVIEW

File: lightwood/encoder/text/pretrained.py [hunk 1]
```diff
@@ -90,46 +94,45 @@ def __init__(
...
         encoded_target_values: torch.Tensor,
     ):
         """
         Fine-tunes a transformer on the priming data.
 
-        CURRENTLY WIP; train + dev are placeholders for a validation-based approach. 
-
-        Train + Dev are concatenated together and a transformer is then fine tuned with weight-decay applied on the transformer parameters. The option to freeze the underlying transformer and only train a linear layer exists if `frozen=True`. This trains faster, with the exception that the performance is often lower than fine-tuning on internal benchmarks.
+        Transformer is fine-tuned with weight-decay on training split. 
+        By default, underlying transformer is frozen and only final linear layer is trained. This trains faster, often as tradeoff for performance.
```
Comment (paxcema) [inline lightwood/encoder/text/pretrained.py:105]: Revert this, as `frozen` default value ended up being unchanged instead.
